bin/kaleidoscope-builder

```diff
@@ -160,37 +159,65 @@ prompt_before_flashing () {
     read
 }
 
+
 flash () {
     maybe_build "$@"
-
+      
     # Check to see if we can see a keyboard bootloader port. 
     # If we -can-, then we should skip over the "reset to bootloader" thing
     find_bootloader_ports
     if [ -z "${DEVICE_PORT_BOOTLOADER}" ]; then
-        prompt_before_flashing
-    
-        # This is defined in the (optional) user config.
-        # shellcheck disable=SC2154
-        ${preFlash_HOOKS}
-    
-        reset_device
-        sleep 2
-        find_bootloader_ports
+        if [ -z "${QMK_FLASH}" ]; then
+            prompt_before_flashing
+          
+            # This is defined in the (optional) user config.
+            # shellcheck disable=SC2154
+            ${preFlash_HOOKS}
+          
+            reset_device
+            sleep 2
+            find_bootloader_ports
+        else
+            # This is defined in the (optional) user config.
+            # shellcheck disable=SC2154
+            ${preFlash_HOOKS}
+        
+            # Keyboards with QMK firmware must be reset manually to enter bootloader mode.
+            echo "Reset your keyboard now to initiate flashing."
+            wait_for_bootloader_port
+        fi
```

Comment:
So, the basic difference here is the manual reset, right? I'd suggest not tieing the name to QMK, because a manual reset can be useful even with Kaleidoscope. So how about we call it `MANUAL_RESET`? And narrow the paths a little, somewhat like his:

```suggestion
        prompt_before_flashing
        # This is defined in the (optional) user config.
        # shellcheck disable=SC2154
        ${preFlash_HOOKS}

        if [ -z "${MANUAL_RESET}" ]; then          
           reset_device
           sleep 2
           find_bootloader_ports
        else
           wait_for_bootloader_port
        fi
```

I do not think we need the extra text, because the pre-flashing prompt already tells us to put the keyboard into flash mode. A manual reset does that.